syntax = "proto3";


message Observation {
  int32 env_id = 1;
  bytes data = 2;
  repeated int32 dimension = 3;
  string dtype = 4;
  float reward = 5;
  bool done = 6;
}


message Action {
  int32 action = 7;
  int32 env_id = 8;
}

// service
service ModelCall {
    rpc MakeAction(stream Observation) returns (stream Action) {}
}